### PR TITLE
Documentation overhaul for converter and gamestate development

### PR DESCRIPTION
**Pull request by heinezen**
Removes outdated parts of the docs and adds new explanations for the current development tasks.

- [x] `README.md`
- [x] `doc/contributing.md`
- [x] `doc/media_convert.md`
- [x] `doc/milestones.md`
- [x] `doc/project_structure.md`
- [x] ~~`doc/status.md`~~ (Removed)
- [x] `doc/troubleshooting.md`
- [x] `doc/code/lzx_compression_info` (convert to Markdown)
- [x] `doc/convert/interactive.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,32 +128,54 @@ or the [bug tracker](https://github.com/SFTtech/openage/issues).
...
+    **Asset Converter:** Converts whatever properietary format used by a Age of Empires 2 into
+    open formats. Written mostly in Python 3. There are a lot of TODOs and beginner issues available
+    right now, so it's a good place to get your feet wet.
+    **Game simulation:** Also known as the gameplay implementation. Written in C++, using the
+    Entity-Component-System paradigm in addition to an event-driven simulation.
+    **Documentation:** We not only document code, but also anything technical about the Genie engine
+    and its games. If you like documenting [file formats](https://github.com/SFTtech/openage/tree/master/doc/media)
+    or thouroughly investigating [game mechanics](https://github.com/SFTtech/openage/tree/master/doc/reverse_engineering),
+    then this might be the right place to start.
+* **Check the [issues** labelled with good first issues](https://github.com/SFTtech/openage/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22). These are tasks that you can start right away
```
Comment (hhirsch) [inline README.md:149]:
```suggestion
* **Check the issues** [labelled with good first issues](https://github.com/SFTtech/openage/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22). These are tasks that you can start right away
```

File: README.md [hunk 2]
```diff
@@ -126,32 +128,54 @@ or the [bug tracker](https://github.com/SFTtech/openage/issues).
...
+  You can look at the project related issues and find something for you, for example:
+    **Asset Converter:** Converts whatever properietary format used by a Age of Empires 2 into
+    open formats. Written mostly in Python 3. There are a lot of TODOs and beginner issues available
+    right now, so it's a good place to get your feet wet.
+    **Game simulation:** Also known as the gameplay implementation. Written in C++, using the
+    Entity-Component-System paradigm in addition to an event-driven simulation.
+    **Documentation:** We not only document code, but also anything technical about the Genie engine
+    and its games. If you like documenting [file formats](https://github.com/SFTtech/openage/tree/master/doc/media)
+    or thouroughly investigating [game mechanics](https://github.com/SFTtech/openage/tree/master/doc/reverse_engineering),
+    then this might be the right place to start.
```
Comment (hhirsch) [inline README.md:148]: So the markdown renderer just puts this into a block which I am sure is not your intention and is hard to read. Maybe make Asset Converter, Game simulation and Documentation a bullet point so it is formated better.

**Review by hhirsch — CHANGES_REQUESTED** [review]
I suggest some format fixes see my other comments.

File: README.md (hunk 1 above)
Comment (heinezen) [reply README.md:149]: good catch :)

File: README.md (hunk 2 above)
Comment (heinezen) [reply README.md]: Yeah I totally forgot to mark the bullet points. Should be fixed now.

**Review by hhirsch — APPROVED** [review]

File: doc/media_convert.md [hunk 3]
```diff
@@ -1,21 +1,35 @@
...
+
+* **RoR**: Age of Empires 1 (1997) + Rise of Rome
+* **AoC**: Age of Empires 2 (1999) + The Conqueror's
+* **SWGB**: Star Wars: Galactic Battlegrounds + Clone Campaigns
+* **DE2**: Age of Empires 2: Definitve Edition
+
+The following versions are currently *unsupported*, but will be added later:
+
+* **HD**: Age of Empires 2 (2013) (formerly: Age of Empires 2: HD Edition)
+* **DE1**: Age of Empires 1: Definitive Edition
```
Comment (hhirsch) [inline doc/media_convert.md:16]: DE 2 conversion is supported and HD isn't? I am not sure about that I had to install HD specifically to get openage to run and I think I was specifically told to do so on IRC. @heinezen

Comment (heinezen) [reply doc/media_convert.md:16]: That's okay. The new converter introduced in #1151 does not support HD yet, but it's probably gonna be implemented back in sooner or later (#1289 ).

File: README.md [hunk 4]
```diff
@@ -16,14 +16,14 @@ To play, you require *an original AoE II: TC or [AoE II: HD](http://store.steamp
...
+Issue Tracker    | [SFTtech/openage](https://github.com/SFTtech/openage/issues)
 Development Blog | [blog.openage.sft.mx](https://blog.openage.sft.mx)
-Forum | [<img src="https://www.redditstatic.com/about/assets/reddit-logo.png" alt="reddit" height="22"/> /r/openage](https://www.reddit.com/r/openage/)
-Matrix Chat | [`#sfttech:matrix.org`](https://riot.im/app/#/room/#sfttech:matrix.org)
-IRC Chat | [`irc.freenode.net #sfttech`](https://webchat.freenode.net/?channels=sfttech)
-Money Sink | [![money sink](https://liberapay.com/assets/widgets/donate.svg)](https://liberapay.com/SFTtech)
+Forum            | [<img src="https://www.redditstatic.com/about/assets/reddit-logo.png" alt="reddit" height="22"/> /r/openage](https://www.reddit.com/r/openage/)
+Matrix Chat      | [`#sfttech:matrix.org`](https://riot.im/app/#/room/#sfttech:matrix.org)
+IRC Chat         | [`irc.freenode.net #sfttech`](https://webchat.freenode.net/?channels=sfttech)
+Money Sink       | [![money sink](https://liberapay.com/assets/widgets/donate.svg)](https://liberapay.com/SFTtech)
```
Comment (TheJJ) [inline README.md:26]: you could change the blog to `blog.openage.dev`.
the spacing around the `|` was chosen like this deliberately in order to not having to reindent all lines once we change one of them :) but it doesn't hurt to align them, dunno.

File: README.md [hunk 5]
```diff
@@ -126,32 +128,54 @@ or the [bug tracker](https://github.com/SFTtech/openage/issues).
...
+  You can look at the project related issues and find something for you, for example:
+    * **Asset Converter:** Converts whatever properietary format used by a Age of Empires 2 into
+    open formats. Written mostly in Python 3. There are a lot of TODOs and beginner issues available
+    right now, so it's a good place to get your feet wet.
+    * **Game simulation:** Also known as the gameplay implementation. Written in C++, using the
+    Entity-Component-System paradigm in addition to an event-driven simulation.
+    * **Documentation:** We not only document code, but also anything technical about the Genie engine
+    and its games. If you like documenting [file formats](https://github.com/SFTtech/openage/tree/master/doc/media)
+    or thouroughly investigating [game mechanics](https://github.com/SFTtech/openage/tree/master/doc/reverse_engineering),
+    then this might be the right place to start.
```
Comment (TheJJ) [inline README.md:148]: links to the tree can be relative, i.e. `doc/media/` etc

File: README.md [hunk 6]
```diff
@@ -126,32 +128,54 @@ or the [bug tracker](https://github.com/SFTtech/openage/issues).
...
+* **Check the issues** [labelled with good first issues](https://github.com/SFTtech/openage/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22). These are tasks that you can start right away
+  and don't require much previous knowledge.
+* **Ask us** in the [chat](https://riot.im/app/#/room/#sfttech:matrix.org). Someone there could need
```
Comment (TheJJ) [inline README.md:151]: it's now `element.io`

File: README.md [hunk 7]
```diff
@@ -126,32 +128,54 @@ or the [bug tracker](https://github.com/SFTtech/openage/issues).
...
+  help with something.
+* You can also **take the initiative** and fix a bug you found, create an issue for discussion or
+  implement a feature that we never though of, but always wanted.
+
+
+## Ok, I found something. What now?
+
+* **[Tell us](#contact)**, if you haven't already. Chances are that we have additional information
+  and directions.
+* [Read the docs](https://github.com/SFTtech/openage/tree/master/doc). They will answer most "administrative"
```
Comment (TheJJ) [inline README.md:161]: also relative link `doc/`

File: README.md [hunk 8]
```diff
@@ -126,32 +128,54 @@ or the [bug tracker](https://github.com/SFTtech/openage/issues).
...
+  questions like what code style is used and how the engine core parts are connected.
+* **Read the code** and get familiar with the engine component you want to work with.
+* Do not hesitate to **[ask us for help](#contact)** if you do not understand something.
+
+
+## How do I contribute my features/changes?
+
+* Read the **[contributing guide](https://github.com/SFTtech/openage/blob/master/doc/contributing.md)**.
```
Comment (TheJJ) [inline README.md:169]: relative link

File: doc/contributing.md [hunk 9]
```diff
@@ -109,7 +108,7 @@ Before making a pull request, it's good to review these things:
 - You should [rebase](#rebasing) your work to avoid "clutter" in your commits and use the latest upstream code.
 
 We have a buildbot (currently [`kevin-ci`](https://github.com/SFTtech/kevin)) that runs all sorts of checks.
-It can be a bit fascist at times, so don't be shocked if it rejects your code, and go fix it instead.
+It can be very strict at times, so don't be shocked if it rejects your code, and go fix it instead.
```
Comment (TheJJ) [inline doc/contributing.md:111]: :grin:

File: doc/contributing.md [hunk 10]
```diff
@@ -205,9 +204,9 @@ git will not accept your pushes because they're not simple fast-forwards:
   - Why? You changed the commits, so you want the old ones to be deleted!
 
   You can use any of:
-  - `git push origin +nobody-likes-you`
-  - `git push origin -f you-are-fat`
-  - `git push origin --force and-you-are-adopted`
+  - `git push origin +my-awesome-feature`
+  - `git push origin -f my-awesome-feature`
+  - `git push origin --force my-awesome-feature`
```
Comment (TheJJ) [inline doc/contributing.md:209]: :rofl:

File: doc/convert/interactive.md [hunk 11]
```diff
@@ -1,5 +1,8 @@
 # Interactive conversion
 
+***Important notice***: This functionality is currently broken. Interactive browsing
+will be reintegrated in a later update.
+
```
Comment (TheJJ) [inline doc/convert/interactive.md:5]: let's hope we don't forget to update this note then :)

File: doc/status.md [hunk 12]
```diff
@@ -1,78 +0,0 @@
...
-#### Buildsystem
-
-Powered mainly by CMake and Python, our [buildsystem](/buildsystem) is dynamically
-rebuilding and regenerating only the files it needs to.
-
-
-#### Job management
-
-Background tasks can be submitted and detached by our [job subsystem](/libopenage/job).
-Used for sound decoding and various other computations.
```
Comment (TheJJ) [inline doc/status.md:78]: we then need to remove it from the website also.

**Review by TheJJ — CHANGES_REQUESTED** [review]
nice

File: README.md (hunk 4 above)
Comment (heinezen) [reply README.md:26]: Done

File: README.md (hunk 6 above)
Comment (heinezen) [reply README.md]: Now fixed everywhere else, too.

File: README.md (hunk 5 above)
Comment (heinezen) [reply README.md:148]: All fixed.

File: README.md (hunk 7 above)
Comment (heinezen) [reply README.md]: fixed

File: README.md (hunk 8 above)
Comment (heinezen) [reply README.md]: fixed

File: doc/convert/interactive.md (hunk 11 above)
Comment (heinezen) [reply doc/convert/interactive.md:5]: The person who updates the Python code hopefully also documents features here and removes the notice :D

File: doc/status.md (hunk 12 above)
Comment (heinezen) [reply doc/status.md:78]: It was already removed when the website relaunched.

File: README.md [hunk 13]
```diff
@@ -9,21 +9,21 @@ To play, you require *an original AoE II: TC or [AoE II: HD](http://store.steamp
...
-Forum | [<img src="https://www.redditstatic.com/about/assets/reddit-logo.png" alt="reddit" height="22"/> /r/openage](https://www.reddit.com/r/openage/)
-Matrix Chat | [`#sfttech:matrix.org`](https://riot.im/app/#/room/#sfttech:matrix.org)
-IRC Chat | [`irc.freenode.net #sfttech`](https://webchat.freenode.net/?channels=sfttech)
-Money Sink | [![money sink](https://liberapay.com/assets/widgets/donate.svg)](https://liberapay.com/SFTtech)
+Contact          | Where?
+-----------------|-------
+Issue Tracker    | [SFTtech/openage](https://github.com/SFTtech/openage/issues)
+Development Blog | [blog.openage.sft.mx](https://blog.openage.dev)
+Forum            | [<img src="https://www.redditstatic.com/about/assets/reddit-logo.png" alt="reddit" height="22"/> /r/openage](https://www.reddit.com/r/openage/)
+Matrix Chat      | [`#sfttech:matrix.org`](https://element.io/app/#/room/#sfttech:matrix.org)
```
Comment (TheJJ) [inline README.md:24]: broken link, think it's https://app.element.io/#/room/#sfttech:matrix.org

**Review by TheJJ — CHANGES_REQUESTED** [review]

**Comment by heinezen** [comment]
@TheJJ All fixed

**Review by TheJJ — APPROVED** [review]
yey

**Pull request merged**